src/ducks/components/App.jsx

```diff
@@ -21,6 +21,17 @@ export class App extends Component {
     props.initApp() // fetch apps without icons
   }
 
+  componentDidUpdate(prevProps) {
+    if (!this.props.location) return
+    // quitting channel modal, so we restore the previous app state
+    if (
+      this.props.location.pathname !== prevProps.location.pathname &&
+      prevProps.location.pathname.match(/.*\/channel\/.*/)
```

Comment:
I thought we were clean with logic based on location ? With our recent changes on routing mechanism ? Maybe I am wrong but if not we need a route here.

Reply:
It's more complicated than just a route here, we need this to detect the action of cancelling an update/channel switching (as discussed)